charts/openstack-cluster/templates/control-plane/kubeadm-control-plane.yaml

```diff
@@ -69,6 +69,61 @@ mounts:
 {{- end }}
 {{- end }}
 
+{{- define "openstack-cluster.controlplane.kubeadmConfigSpec.etcdEncryption" -}}
+{{- if .Values.etcd.encryption.enabled }}
+preKubeadmCommands:
+  - mkdir -p /etc/kubernetes/enc
+  - mkdir -p /etc/kubernetes/patches
+initConfiguration:
+  patches:
+    directory: /etc/kubernetes/patches
+joinConfiguration:
+  patches:
+    directory: /etc/kubernetes/patches
+clusterConfiguration:
+  apiServer:
+    extraArgs:
+      encryption-provider-config: /etc/kubernetes/enc/enc.yaml
+files:
+  - path: /etc/kubernetes/patches/kube-apiserver1+strategic.yaml
+    permissions: "0644"
+    owner: root:root
+    content: |
+      spec:
+        containers:
+        - name: kube-apiserver
+          volumeMounts:
+          - mountPath: /etc/kubernetes/enc
+            name: kube-enc
+            readOnly: true
+        volumes:
+        - hostPath:
+            path: /etc/kubernetes/enc
+            type: DirectoryOrCreate
+          name: kube-enc
+  - path: /etc/kubernetes/enc/enc.yaml
+    content: |
+      apiVersion: apiserver.config.k8s.io/v1
+      kind: EncryptionConfiguration
+      resources:
+        - resources:
+            {{- .Values.etcd.encryption.resources | toYaml | nindent 10 }}
+          providers:
+            - {{ .Values.etcd.encryption.provider }}:
+                keys:
+                  - name: key1
+                    {{- $secret := (lookup "v1" "Secret" .Release.Namespace (print .Release.Name "-etcd-key")) -}}
+                    {{- if $secret }}
+                    secret: {{ $secret.data.key }}
+                    {{- else }}
+                    secret: "Fallback value for CI only"
```

Comment:
This seems a bit dangerous, can we fix up CI so it has a secret created that can be used?

Reply:
Lets add the context in a comment if we can, that sounds like a good call.